components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/model/AuthenticationRequestUser.java

```diff
@@ -0,0 +1,168 @@
+import org.wso2.carbon.identity.application.authenticator.adapter.AuthenticatorAdapterConstants;
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.common.AbstractUserStoreManager;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class holds the authenticated user object which is communicated to the external authentication service.
+ */
+public class AuthenticationRequestUser extends User {
+
+    private String idp;
+    private String sub;
+    private String externalId;
+    private final List<UserClaim> userClaims =  new ArrayList<>();
+
+    public AuthenticationRequestUser(String id){
+
+        super(id);
+    }
+
+    public AuthenticationRequestUser(String id, AuthenticatedUser user) {
+
+        super(id);
+        sub = user.getAuthenticatedSubjectIdentifier();
+
+        Map<ClaimMapping, String> userAttributes = user.getUserAttributes();
+        if (userAttributes != null) {
+            for (ClaimMapping claimMap : userAttributes.keySet()) {
+                String claimUri = claimMap.getLocalClaim().getClaimUri();
+                if (AuthenticatorAdapterConstants.AuthRequestEntityPaths.EXTERNAL_ID_CLAIM.equals(claimUri)) {
+                    externalId = userAttributes.get(claimMap);
+                    break;
+                }
+                userClaims.add(new UserClaim(claimUri, userAttributes.get(claimMap)));
+            }
+        }
+    }
+
+    public void setIdp(String idp) {
+        this.idp = idp;
+    }
+
+    public void setSub(String sub) {
+        this.sub = sub;
+    }
+
+    public void setExternalId(String externalId) {
+        this.externalId = externalId;
+    }
+
+    public void setUserClaims(UserClaim userClaim) {
+        userClaims.add(userClaim);
+    }
+
+    /**
+     * Validate and return userType. The user type has to be either LOCAL or FED, if it is not any of them or
+     * not provided throw as exception.
+     *
+     * @return True if the idp is FED, false if idp is LOCAL.
+     * @throws ActionExecutionResponseProcessorException If invalid user type is provided.
+     */
+    public boolean resolveUserType()
+            throws ActionExecutionResponseProcessorException {
+
+        if (StringUtils.equalsIgnoreCase(
+                AuthenticatorAdapterConstants.AuthRequestEntityPaths.UserType.LOCAL.toString(), idp)) {
+            return false;
+        } else if (StringUtils.equalsIgnoreCase(
+                AuthenticatorAdapterConstants.AuthRequestEntityPaths.UserType.FED.toString(), idp)) {
+            return true;
+        } else {
+            throw new ActionExecutionResponseProcessorException("Undefined user idp type is provided for the " +
+                    "authenticated user in external authenticator service response payload: " + idp);
+        }
```

Comment:
These needs to be incorporated for troubleshooting logs

Reply:
Fixed with commit e35699ea008a4b1df7251ac6162b6e9634aacf77